### ISSUE TITLE
Reactive handler without output

### ISSUE THREAD

**Issue by MichalMichalak** [spring-cloud/spring-cloud-stream#1583]
Is there a way to write a reactive handler without output specified? I've checked the docs and they don't mention anything like that. Sure there is an option available with `@StreamListener` when you have both - input and output, but nothing with input only. There is question opened on SO[1] but no one seem to be interested in it. I've also tried to search the net but nothing meaningful emerged. I know this may not be the place to open "how to" issue, but I didn't know where else to go.

[1] https://stackoverflow.com/questions/54031061/spring-cloud-stream-reactive-listener-without-output

**Comment by olegz** [comment]
@MichalMichalak In the future please use [Stack Overflow channel](https://stackoverflow.com/questions/tagged/spring-cloud-stream) for asking question. We generally use GH ***only*** for raising issues.
That said, have you tried accomplishing what you're looking for via [Spring Cloud Function support](https://spring.io/blog/2018/08/28/spring-cloud-stream-fishtown-m2-2-1-0-m2-release-announcement) that was recently added and is available in Fishtown.RELEASE (2.1.0).

Basically if I understand correctly all you're looking for something like this:

```
@SpringBootApplication
@EnableBinding(Sink.class)
public class MyFunctionBootApp {

	public static void main(String[] args) {
		SpringApplication.run(MyFunctionBootApp.class, 
                  "--spring.cloud.stream.function.definition=noOutputConsumer");
	}

	@Bean
	public Consumer<Flux<String>> noOutputConsumer() {
		return flux -> flux.subscribe(System.out::println);
	}
}
```

**Comment by MichalMichalak** [comment]
Thank you for answer and sorry for opening issue here. I did it because proposed solution (which is also there, on SO) seems at least strange to me. What is the benefit on using SCF over SCS here if you must subscribe by yourself? I was under the impression that subscribing by yourself is generally discouraged. I was expecting rather to get `Flux`, do the job and pass it to the framework (return the flux from my function.) As it is now, I can easily use `@StreamListener` with input only and effect would be the same. Unless, there is something more to it.

**Comment by olegz** [comment]
Sure, in that case you can do this:
```
@Bean
public Function<Flux<String>, Mono<Void>> noOutputConsumer() {
	return flux -> flux.map(..);
}
```

The issue with Consumer is that by definition (not by us) it returns nothing, so there is nothing for us to trigger, hence the user has to trigger it themselves.

**Comment by olegz** [comment]
I am not sure what exception your are referring to, so can't comment.
Also, with regard to _...let me get back to SCSt for a moment..._. We are talking SCSt, we never left ;)
Spring Cloud Function (SCF) is the [new programming model supported by SCSt](https://spring.io/blog/2018/10/30/spring-cloud-stream-fishtown-rc1-2-1-0-rc1-release-announcement), and given that Flux is the lowest denominator in SCF it makes reactive module and all the code in it redundant to the point where we are planning it's [end of days](https://github.com/spring-cloud/spring-cloud-stream/issues/1559)

**Comment by olegz** [comment]
Hey, no problem at all. It was a good discussion that was worth to be discussed here as it helped us to better understand how you are using SCSt
